R/utils.R

```diff
@@ -55,3 +92,89 @@ str_standardise <- function(name) {
 str_trim_ext <- function(path) {
   sub("\\..+$", "", path)
 }
+
+gather_tree <- function(x, depth) {
```

Comment:
@hadley just wrote this `gather_tree()` util that may be useful in purrr or tidyr.

It takes a tree of data frames and a depth and gather the data frames together. `depth` can be a character vector whose elements are passed to `.id` in `bind_rows()`.

It's accompanied with `spread_attributes()` which creates a data frame out of an atomic vector and its attributes (which should be as long as the vector).

There is also `at_bottom()` which I use to map `spread_attributes()` on a tree of vectors + metadata to create data frames at the bottom. I think that's altogether a nice little idiom.